### PR TITLE
admin button to auto-gen & link Discord roles

### PR DESCRIPTION
**Pull request by npinsker**
image

addresses #671 

- Button on the sidebar that's visible to admins only
- Clicking it will generate Discord roles for all tags that are blue or white
- Also creates a `ChatRole` for all Discord roles, meaning people with those roles will get pinged when a puzzle's tagged with the corresponding tag

### CODE REVIEW

**Comment by rgossiaux** [comment]
Why did you move this from the hunt edit page to the sidebar? I thought that made the most sense to me.

I also think it's unlikely we ever want to modify these during the hunt FWIW

**Comment by npinsker** [comment]
moved back to `/edit`

File: chat/tasks.py [hunk 1]
```diff
@@ -166,3 +166,56 @@ def handle_sheet_created(puzzle_id):
...
+            )
+            discord_roles_by_name[tag.name] = new_role_info
+            logger.info(f"Created new Discord role {tag.name}")
+
+        # Copy tag info into a ChatRole
+        existing_chat_role = existing_chat_roles.filter(name=tag.name)
+        if existing_chat_role.exists():
+            obj = existing_chat_role.first()
+            if obj.role_id != discord_roles_by_name[tag.name]["id"]:
+                with transaction.atomic():
```
Comment (rgossiaux) [inline chat/tasks.py:211]: Does this actually do anything if we're just doing a single operation? ditto below

File: hunts/views.py [hunk 2]
```diff
@@ -144,6 +144,26 @@ def redirect_to_drive(request, hunt_slug):
...
+        return HttpResponseForbidden()
+
+    hunt = Hunt.get_object_or_404(user=request.user, slug=hunt_slug)
+
+    import chat.tasks
+
+    chat.tasks.sync_roles(
+        hunt,
+        settings.CHAT_DEFAULT_SERVICE,
+    )
```
Comment (rgossiaux) [inline hunts/views.py:161]: I think this will run this synchronously by the web process, right? Need to call `.delay()` if you want to create a task

**Review by rgossiaux — APPROVED** [review]
Lgtm, nice!

**Pull request merged**